merlin/models/tf/models/base.py

```diff
@@ -284,6 +285,18 @@ def _set_save_spec(self, inputs, args=None, kwargs=None):
 
 
 class BaseModel(tf.keras.Model):
+    def __init__(self, **kwargs):
+        super(BaseModel, self).__init__(**kwargs)
+
+        # Initializing model control flags controlled by MetricsComputeCallback()
+        self._should_compute_train_metrics_for_batch = tf.Variable(
```

Comment:
This needs to be moved to the `__init__` method (from the compile) method so that this variable is created when reloading the model.